From 4a7b1bd1a70086d3b6839a43e583b287eab22c30 Mon Sep 17 00:00:00 2001
From: Praveen Diwakar <praveen.diwakar@intel.com>
Date: Thu, 9 Apr 2015 23:00:01 +0530
Subject: [PATCH] Audio: Disable micbias when not in use

Disable micbias pin when no audio use case is running

Change-Id: I3bc0036771bb611bb91d0688485418e3bb50ca7f
Tracked-On: https://jira01.devtools.intel.com/browse/GMINL-8137
Signed-off-by: Praveen Diwakar <praveen.diwakar@intel.com>
---
 sound/soc/intel/board/cht_cr_aic3100.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/sound/soc/intel/board/cht_cr_aic3100.c b/sound/soc/intel/board/cht_cr_aic3100.c
index 3efbda2..6e9d000 100644
--- a/sound/soc/intel/board/cht_cr_aic3100.c
+++ b/sound/soc/intel/board/cht_cr_aic3100.c
@@ -99,7 +99,7 @@ static inline void cht_set_mic_bias(struct snd_soc_codec *codec, bool enable)
 	if (enable)
 		cht_force_enable_pin(codec, "micbias", true);
 	else
-		cht_force_enable_pin(codec, "micbias", true);
+		cht_force_enable_pin(codec, "micbias", false);
 	snd_soc_dapm_sync(&codec->dapm);
 }
 
@@ -717,7 +717,6 @@ static int cht_audio_init(struct snd_soc_pcm_runtime *runtime)
 		return ret;
 	}
 
-	cht_set_mic_bias(codec, true);
 	ret = snd_soc_dapm_sync(&card->dapm);
 	if (ret) {
 		pr_err("unable to sync dapm\n");
-- 
1.9.1

